bughouse_console/src/rusqlite_server_hooks.rs

```diff
@@ -0,0 +1,183 @@
+use bughouse_chess::event::*;
+use bughouse_chess::server::*;
+use bughouse_chess::server_hooks::ServerHooks;
+use bughouse_chess::*;
+use log::error;
+
+pub struct RusqliteServerHooks {
+    invocation_id: String,
+    game_number: u64,
+    game_start_time: Option<std::time::SystemTime>,
+    conn: rusqlite::Connection,
+}
+
+impl RusqliteServerHooks {
+    pub fn new(address: &str) -> Result<Self, Box<dyn std::error::Error>> {
+        let conn = rusqlite::Connection::open(address)?;
+        conn.execute(
+            "CREATE TABLE IF NOT EXISTS finished_games (
+                    invocation_id TEXT NOT NULL,
+                    game_number INTEGER NOT NULL,
+                    game_start_time TEXT,
+                    game_end_time TEXT,
+                    player_red_a TEXT,
+                    player_red_b TEXT,
+                    player_blue_a TEXT,
+                    player_blue_b TEXT,
+                    result TEXT,
+                    PRIMARY KEY(invocation_id, game_number))",
+            (),
+        )?;
+        Ok(Self {
+            invocation_id: uuid::Uuid::new_v4().to_string(),
+            game_number: 1,
+            game_start_time: None,
+            conn: conn,
+        })
+    }
+
+    pub fn prepare_database(address: &str) -> Option<()> {
+        None
+    }
+}
+
+impl ServerHooks for RusqliteServerHooks {
+    fn on_event(
+        &mut self,
+        event: &BughouseServerEvent,
+        maybe_game: Option<&GameState>,
+        round: usize,
+    ) {
+        if let BughouseServerEvent::GameStarted { .. } = event {
+            self.game_start_time = Some(std::time::SystemTime::now());
```

Comment:
Let's use UTC time for unambiguity

Reply:
Done. I'm now formatting as rfc3339, not sure what's best. That's human-readable but selecting a time range for example isn't too robust. Though if we only want a specific date, selecting that data is trivial as a prefix of the string.

I wouldn't mind this being a seconds or microseconds integer timestamp instead.